cmd/thanos/receive.go

```diff
@@ -226,9 +231,21 @@ func runReceive(
 	allowOutOfOrderUpload bool,
 	comp component.SourceStoreAPI,
 	hashFunc metadata.HashFunc,
+	receiverMode receive.ReceiverMode,
 ) error {
+
 	logger = log.With(logger, "component", "receive")
 	level.Warn(logger).Log("msg", "setting up receive")
+
+	switch receiverMode {
+	case receive.INGESTOR_MODE:
```

Comment:
I wonder if we can use this time group together distributing work and ingesting mode, so it's selected in one time. (:

Reply:
let's start with separate functions to register all?

Reply:
I tried using functions to separate the logic but seems like the functions convoluted the codebase, and passing a large list of parameters was distracting to read.

I have tried collecting the similar sections into one `if` condition, let me know if this makes sense (: